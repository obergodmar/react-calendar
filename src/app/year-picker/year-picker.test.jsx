import React from "react";
import { shallow } from "enzyme";
import { describe, jest, test } from "@jest/globals";

import { YearPicker } from ".";

describe("Year Picker", () => {
  const handleAddYear = jest.fn();
  const date = 1970;
  const wrapper = shallow(<YearPicker addYear={handleAddYear} date={date} />);

  test("Should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should decrease year by one", () => {
    wrapper.find(".left").simulate("click");
    expect(handleAddYear).toHaveBeenCalledWith(-1);
  });

  test("Should increase year by one", () => {
    wrapper.find(".right").simulate("click");
    expect(handleAddYear).toHaveBeenCalledWith(1);
  });

  test("Should decrease year by 25", () => {
    wrapper.find(".doubleLeft").simulate("click");
    expect(handleAddYear).toHaveBeenCalledWith(-25);
  });

  test("Should increase year by 25", () => {
    wrapper.find(".doubleRight").simulate("click");
    expect(handleAddYear).toHaveBeenCalledWith(25);
  });
});
