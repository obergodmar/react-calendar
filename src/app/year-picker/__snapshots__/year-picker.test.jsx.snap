// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Year Picker Should match snapshot 1`] = `
<div
  className="yearPicker"
>
  <button
    className="button doubleLeft"
    onClick={[Function]}
  />
  <button
    className="button left"
    onClick={[Function]}
  />
  <span>
    1970
  </span>
  <button
    className="button right"
    onClick={[Function]}
  />
  <button
    className="button doubleRight"
    onClick={[Function]}
  />
</div>
`;
